packages/x-tree-view/src/internals/plugins/useTreeViewExpansion/useTreeViewExpansion.test.tsx

```diff
@@ -0,0 +1,244 @@
+      const onExpandedItemsChange = spy();
+
+      const response = render({
+        items: [{ id: '1', children: [{ id: '1.1' }] }],
+        onExpandedItemsChange,
+      });
+
+      fireEvent.click(response.getItemContent('1'));
+      act(() => {
+        response.getRoot().focus();
+      });
+
+      expect(onExpandedItemsChange.callCount).to.equal(1);
+      expect(onExpandedItemsChange.lastCall.args[1]).to.deep.equal(['1']);
+    });
+
+    it('should call the onExpandedItemsChange callback when the model is updated (add expanded item no non-empty list)', () => {
+      const onExpandedItemsChange = spy();
+
+      const response = render({
+        items: [
+          { id: '1', children: [{ id: '1.1' }] },
+          { id: '2', children: [{ id: '2.1' }] },
+        ],
+        onExpandedItemsChange,
+        defaultExpandedItems: ['1'],
+      });
+
+      fireEvent.click(response.getItemContent('2'));
+      act(() => {
+        response.getRoot().focus();
+      });
+
+      expect(onExpandedItemsChange.callCount).to.equal(1);
+      expect(onExpandedItemsChange.lastCall.args[1]).to.deep.equal(['2', '1']);
+    });
+
+    it('should call the onExpandedItemsChange callback when the model is updated (remove expanded item)', () => {
+      const onExpandedItemsChange = spy();
+
+      const response = render({
+        items: [
+          { id: '1', children: [{ id: '1.1' }] },
+          { id: '2', children: [{ id: '2.1' }] },
+        ],
+        onExpandedItemsChange,
+        defaultExpandedItems: ['1'],
+      });
+
+      fireEvent.click(response.getItemContent('1'));
+      act(() => {
+        response.getRoot().focus();
+      });
+
+      expect(onExpandedItemsChange.callCount).to.equal(1);
+      expect(onExpandedItemsChange.lastCall.args[1]).to.deep.equal([]);
+    });
+
+    it('should warn when switching from controlled to uncontrolled', () => {
+      const response = render({
+        items: [{ id: '1' }],
+        expandedItems: [],
+      });
+
+      expect(() => {
+        response.setProps({ expandedItems: undefined });
+      }).toErrorDev(
+        'MUI X: A component is changing the controlled expandedItems state of TreeView to be uncontrolled.',
+      );
+    });
+
+    it('should warn and not react to update when updating the default state', () => {
+      const response = render({
+        items: [{ id: '1', children: [{ id: '1.1' }] }, { id: '2' }],
+        defaultExpandedItems: ['1'],
+      });
+
+      expect(() => {
+        response.setProps({ defaultExpandedItems: ['2'] });
+        expect(response.isItemExpanded('1')).to.equal(true);
+        expect(response.isItemExpanded('2')).to.equal(false);
+      }).toErrorDev(
+        'MUI X: A component is changing the default expandedItems state of an uncontrolled TreeView after being initialized. To suppress this warning opt to use a controlled TreeView.',
+      );
+    });
+  });
+
+  describe('click interactions', () => {
```

Comment:
Before we tested click on the label and the icon, I added tests for the content directly because when customizing, people might have other elements and clicking those should still toggle expansion.